FHEM/14_SD_WS.pm

```diff
@@ -125,6 +127,49 @@ sub SD_WS_Undef {
   my ($hash, $name) = @_;
   carp "SD_WS_Undef, too few arguments ($hash, $name)" if @_ < 2;
   delete($modules{SD_WS}{defptr}{$hash->{CODE}}) if(defined($hash->{CODE}) && defined($modules{SD_WS}{defptr}{$hash->{CODE}}));
+	RemoveInternalTimer($hash);
+  return;
+}
+
+#############################
+sub SD_WS_Set {
+  my ($hash, $name, $cmd, @args) = @_;
+  my $ret = undef;
+	my $LastInputDev = $hash->{LASTInputDev};
+  return $ret if (!defined $LastInputDev);
+	
+  my $rhash = $defs{$LastInputDev};
+  my $rlongids = AttrVal($rhash->{NAME},'longids',0);
+	my $def = $hash->{DEF};
+	my @array=split(/_/,$def);
+	my $model = $array[0] . '_' . $array[1] . '_' . $array[2];
+
+  if (($rlongids ne "0") && ($rlongids eq "1" || $rlongids eq "ALL" || ($rlongids =~ m/$model/))) {
+    if ($cmd eq '?') {
+      $ret = 'replaceBatteryForSec'; # return setList
+    } elsif ($cmd eq 'replaceBatteryForSec') {
+      return "$name: No or too small value specified for the replaceBatteryForSec command" if (int @args < 1 || $args[0] < 1);
+      my @found = devspec2array("TYPE=SD_WS:FILTER=replaceBattery>0");
```

Comment:
Ich habe den Filter angepasst, so dass nur in Internals gesucht wird.
Allerdings könnte ich mir vorstellen, dass das Internal ggf. nicht sichtbar sein sollte?

```suggestion
      my @found = devspec2array("TYPE=SD_WS:FILTER=i:replaceBattery>0");
```

Reply:
OK